### ISSUE TITLE
Missing inclusions of <unistd.h> cause implicit declaration of function errors

### ISSUE THREAD

**Issue by ryandesign** [phmarek/fsvs#24]
You've forgotten to `#include <unistd.h>` for the declaration of `sleep` in add_unvers.c and cp_mv.c and for the declarations of `sleep` and `unlink` in hash_ops.c and resolve.c:

```
add_unvers.c:164:2: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        BUG_ON( action->i_val != RF_UNVERSION && action->i_val != RF_ADD );
        ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
1 error generated.
make[1]: *** [add_unvers.o] Error 1
make: *** [all] Error 2

...

cp_mv.c:580:5: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                                BUG_ON(simil_dir_count > sizeof(similar_dirs)/sizeof(similar_dirs[0]));
                                ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
cp_mv.c:651:2: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        BUG_ON(cand->match_count > 1000 || cand->match_count < 0);
        ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
cp_mv.c:711:2: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        BUG_ON(sizeof(candidates[0].matches_where) *4 < CM___MATCH_NUM,
        ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
cp_mv.c:758:4: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                        BUG_ON(candidate_count > sizeof(candidates)/sizeof(candidates[0]));
                        ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
cp_mv.c:1052:2: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        BUG_ON(used >= buflen);
        ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
cp_mv.c:1077:2: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        BUG_ON(len > cache->entries[cache->lru]->len);
        ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
cp_mv.c:1250:3: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                BUG_ON(1, "fetch list of entries from the repository");
                ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
cp_mv.c:1595:3: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                BUG_ON(!url);
                ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
cp_mv.c:1632:2: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        BUG_ON(!copied, "Copy-sub but no base?");
        ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
9 errors generated.
make[1]: *** [cp_mv.o] Error 1

...

hash_ops.c:221:21: error: implicit declaration of function 'unlink' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                STOPIF_CODE_ERR( (unlink(cp) == -1) && (errno != ENOENT), errno,
                                  ^
hash_ops.c:237:21: error: implicit declaration of function 'unlink' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                        STOPIF_CODE_ERR( unlink(cp) == -1, errno,
                                         ^
hash_ops.c:528:3: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                BUG_ON(listd.dsize != sizeof(list));
                ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
hash_ops.c:607:2: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        BUG_ON( c<=0 || c>HASH__LIST_MAX, "number of entries=%d", c);
        ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
4 errors generated.
make[1]: *** [hash_ops.o] Error 1

...

resolve.c:150:3: error: implicit declaration of function 'sleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                BUG_ON(to_remove - mapped > st.size);
                ^
./global.h:818:42: note: expanded from macro 'BUG_ON'
#define BUG_ON(cond, ...) do { if (cond) BUG( "INTERNAL BUG\n  " #cond "\n  " __VA_ARGS__); } while (0)
                                         ^
./global.h:808:101: note: expanded from macro 'BUG'
#define BUG(...) do { fflush(NULL); debuglevel=1; DEBUGP(__VA_ARGS__); *(int*)42=__LINE__; while(1) sleep(1); } while (0)
                                                                                                    ^
resolve.c:152:7: error: implicit declaration of function 'unlink' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                if (unlink(to_remove) == -1)
                    ^
2 errors generated.
make[1]: *** [resolve.o] Error 1
```

**Comment by ryandesign** [comment]
The fix:

```diff
--- src/add_unvers.c.orig	2023-12-13 06:06:55.000000000 -0600
+++ src/add_unvers.c	2024-07-18 00:00:35.000000000 -0500
@@ -7,6 +7,7 @@
  ************************************************************************/
 
 #include <fcntl.h>
+#include <unistd.h>
 
 #include "global.h"
 #include "add_unvers.h"
--- src/cp_mv.c.orig	2023-12-13 06:06:55.000000000 -0600
+++ src/cp_mv.c	2024-07-18 00:02:28.000000000 -0500
@@ -8,6 +8,7 @@
 
 #include <fcntl.h>
 #include <search.h>
+#include <unistd.h>
 
 #include "global.h"
 #include "cp_mv.h"
--- src/hash_ops.c.orig	2023-12-13 06:06:55.000000000 -0600
+++ src/hash_ops.c	2024-07-18 00:07:31.000000000 -0500
@@ -9,6 +9,7 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <string.h>
+#include <unistd.h>
 
 #include "global.h"
 #include "waa.h"
--- src/resolve.c.orig	2023-12-13 06:06:55.000000000 -0600
+++ src/resolve.c	2024-07-18 00:10:18.000000000 -0500
@@ -9,6 +9,7 @@
 #include <fcntl.h>
 #include <sys/uio.h>
 #include <sys/mman.h>
+#include <unistd.h>
 
 
 #include "global.h"
```

**Comment by phmarek** [comment]
Fixed via 320d7b4dfc3fec5f63fadd962de38df4b3c67e81.
